hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngineImpl.java

```diff
@@ -135,14 +137,35 @@ public NodeEngineImpl(final Node node) {
                 operationService.getAsyncResponseHandler(),
                 operationService.getInvocationMonitor(),
                 eventService,
-                new ConnectionManagerPacketHandler());
+                new ConnectionManagerPacketHandler(),
+                newJetPacketHandler());
         this.quorumService = new QuorumServiceImpl(this);
         this.diagnostics = newDiagnostics();
 
         serviceManager.registerService(InternalOperationService.SERVICE_NAME, operationService);
         serviceManager.registerService(OperationParker.SERVICE_NAME, operationParker);
     }
 
+    private PacketHandler newJetPacketHandler() {
+        // currently service registration is done after the creation of the packet dispatcher, hence
+        // we need to lazily initialize the jet packet handler
+        return new PacketHandler() {
+
+            private volatile PacketHandler handler;
+
+            @Override
+            public void handle(Packet packet) throws Exception {
+                if (handler == null) {
+                    handler = serviceManager.getService(JET_SERVICE_NAME);
+                    if (handler == null) {
+                        throw new UnsupportedOperationException("Jet is not registered on this node");
```

Comment:
at the moment jet packets enter the system and you have no appropriate handler.. what do you want to do otherwise?